aiida/tools/importexport/dbexport/zip.py

```diff
@@ -127,23 +136,18 @@ def insert_path(self, src, dest_name=None, overwrite=True):
         if not os.path.isabs(src):
             raise ValueError('src must be an absolute path in insert_file')
 
-        if not overwrite:
-            try:
-                self._zipfile.getinfo(base_filename)
-                exists = True
-            except KeyError:
-                exists = False
-            if exists:
-                raise IOError('destination already exists: {}'.format(base_filename))
+        if not overwrite and self.exists(base_filename):
+            raise IOError('destination already exists: {}'.format(base_filename))
 
         # print src, filename
         if os.path.isdir(src):
             for dirpath, dirnames, filenames in os.walk(src):
                 relpath = os.path.relpath(dirpath, src)
                 if not dirnames and not filenames:
                     real_src = dirpath
-                    real_dest = os.path.join(base_filename, relpath)
-                    self._zipfile.write(real_src, real_dest)
+                    real_dest = os.path.join(base_filename, relpath) + '/'
```

Comment:
Why the trailing slash? If this is really necessary, one should use the `os.path.sep`

Reply:
Cheers. Well, it seems the slash was really the issue all along here, since _with_ the slash, the string now represents a possible existing path in the zip-file. This is due to zip adding an ending slash to signify a folder type.